exasol_data_science_utils_python/model_utils/reservoir_shuffle.py

```diff
@@ -0,0 +1,109 @@
+from typing import Iterator
+
+import numpy as np
+import pandas as pd
+
+
+class ReservoirShuffle:
+    def __init__(self, it: Iterator[pd.DataFrame], buffer_size: int, batch_size: int):
+        self.it = it
+        self.fill_level = 0
+        self.internal_buffer_size = buffer_size + batch_size
+        self.free_bitset = np.ones(shape=self.internal_buffer_size, dtype=bool)
+        self.buffer_df = None
+        self.batch_size = batch_size
+        self.buffer_size = buffer_size
+
+    def df_empty(self, columns, dtypes, size: int):
+        assert len(columns) == len(dtypes)
+        df = pd.DataFrame(index=range(size))
+        for c, d in zip(columns, dtypes):
+            df[c] = pd.Series(np.zeros(shape=size, dtype=d))
+        return df
```

Comment:
Does it really need to be a class member? You don't use _self_ here; thus the client will need to instantiate an object of ReservoirShuffle(), but maybe only wants to use this method. Maybe you could isolate into a free function.